python/requirements_autoscaler.txt

```diff
@@ -1,28 +1,20 @@
-gym[atari]
-opencv-python-headless
-tensorflow
-lz4
-keras
-smart_open
-tensorflow_probability
-h5py
 bayesian-optimization
-hyperopt==0.1.2
 ConfigSpace==0.4.10
-sigopt
-nevergrad 
-scikit-optimize 
+google-api-python-client
+google-oauth
+h5py
 hpbandster 
+hyperopt==0.1.2
+ipython
+keras
 lightgbm 
-xgboost 
+mlflow
+nevergrad 
+oauth2client
+scikit-optimize 
+sigopt
+smart_open
+tensorflow_probability
```

Comment:
@barakmich thanks! The way we're installing dependencies right now is a bit of a mess. Can you explain the semantics of these files? Most of these dependencies are only used in the tests. So maybe this should be called `requirements_test_autoscaler.txt` or something like that? Same with RLlib, Tune, etc.

Also, maybe these should just be the dependencies that are required for the tests that are not already included in `setup.py` (instead of the union).

All dependencies that users need are listed in `setup.py`.

Basically, it'd be helpful to see a clear explanation of what the "meaning" of each requirements.txt file is and whether it is relevant for users or just testing or something else.

Reply:
Yeah, that tracks. With the setup.py trick I'm about to try from the comments above, and a little guide, it should be doable.

The reason I kept setup.py separate initially is exactly that -- all the dependencies /users/ need are listed there. That's all for a standard user.

The rest are for developers of the different sub-pieces (and tests, though I didn't know that) -- requirements.txt in general being aimed at the developer-of-ray experience. So yeah, some of these can be combined for sure, if that makes sense. I just broke them out by specific use-case from the tooling (where they were repeated and often out of sync). The good news is that follow up PRs can massage these files and not have to change them in many different places.